sagemaker/launch/realtime_entry_points/node_prediction_entry.py

```diff
@@ -0,0 +1,386 @@
+    request_content_type: str
+        String to indicate what is the format of the payload. For GraphStorm built-in real-time
+        input function, the format should be 'application/json'.
+
+    Return
+    -------
+    dgl_graph: DGLGraph
+        A DGL graph for inference. For GraphStorm built-in inference pipeline, the graph is a
+        `DGLGraph` instance.
+    target_dict: 
+    """
+
+    logging.info('-- START processing input data... ')
+    s_t = dt.now()
+
+    logging.debug(request_body)
+
+    try:
+        payload_data = json.loads(request_body)
+    except Exception as e:
+        res = res_msg.json_format_error(error=e)
+        return res
+
+    # TODO(Jian), build a unified payload content sanity checking  method under gconstruct package
+    # to be shared by all entry point files
+
+    # the version object will be used later to keep backward compatibilty for early versions
+    version = payload_data.get('version', None)
+
+    gml_task = payload_data.get('gml_task', None)
+    targets = payload_data.get('targets', None)
+
+    # 1. check if the payload is for a NC task
+    if gml_task is None or (gml_task not in ['node_classification', 'node_regression']):
+        track = f'This endpoint is for node prediction, but got {gml_task} task from the payload.'
+        res = res_msg.model_mismatch_error(track=track)
+        return res
+
+    # 2. check if the targets field is provided
+    if targets is None or len(targets)==0:
+        res = res_msg.missing_required_field(field='targets')
+        return res
+    # check if target has node_type and node_id. Will check values in id mapping
+    for target in targets:
+        if isinstance(target, dict):
+            if 'node_type' not in target or 'node_id' not in target:
+                res = res_msg.json_format_error(error=('The Element of \"targets\" field should ' \
+                    'be a dictionary that has both \"node_type\" and \"node_id\" keys, but ' \
+                    f'got {target}.'))
+                return res
+        else:
+            res = res_msg.json_format_error(error=('The Element of \"targets\" field should ' \
+                f'be a dictionary, but got {target}'))
+            return res
+
+    # processing payload to generate a DGL graph
+    global CONFIG_JSON
+    g_resp = process_json_payload_graph(payload_data, CONFIG_JSON)
+
+    if g_resp[STATUS] == 400:
+        track = f'Error code: {g_resp[ERROR_CODE]}, Message: {g_resp[MSG]}.'
+        res = res_msg.graph_construction_failure(track=track)
+        return res
+
+    # succeeded
+    if g_resp[STATUS] == 200:
+        dgl_graph = g_resp[GRAPH]
+        raw_node_id_maps = g_resp[NODE_MAPPING]
+
+    # mapping the targets, a list of node objects, to new graph node IDs after dgl graph
+    # construction for less overall data processing time
+    target_mapping_dict = {}
+    for target in targets:
+        target_ntype = target['node_type']
+        target_nid = target['node_id']
```

Comment:
Is target_nid a string or a list?

Reply:
As the API specification defined, the `target_nid` is a string.